addons/delivery/models/sale_order.py

```diff
@@ -96,12 +98,12 @@ def _create_delivery_line(self, carrier, price_unit):
             taxes_ids = self.fiscal_position_id.map_tax(taxes, carrier.product_id, self.partner_id).ids
 
         # Create the sales order line
-        carrier_with_partner_lang = carrier.with_context(lang=self.partner_id.lang)
-        if carrier_with_partner_lang.product_id.description_sale:
-            so_description = '%s: %s' % (carrier_with_partner_lang.name,
-                                        carrier_with_partner_lang.product_id.description_sale)
+
+        if carrier.product_id.description_sale:
```

Comment:
I guess the idea here is only to clean the code (because this diff does not change the result). But in that case, I think we could also remove the line `carrier = carrier.with_context(lang=self.partner_id.lang)`
However, it's often better to minimize the diff of a fix. Considering this, I think we could let the code as it is

Reply:
i think both `carrier = carrier.with_context` (to translate fields called on the recordset with the context) and `context = {` (to translate `_('...')`  in current method) are necessary